metrics-operator/controllers/common/providers/dynatrace/dynatrace.go

```diff
@@ -74,13 +78,23 @@ func (d *KeptnDynatraceProvider) EvaluateQuery(ctx context.Context, metric metri
 	err = json.Unmarshal(b, &result)
 	if err != nil {
 		d.Log.Error(err, "Error while parsing response")
-		return "", nil, err
+		return "", b, err
+	}
+	if !reflect.DeepEqual(result.Error, Error{}) {
+		err = fmt.Errorf(ErrAPI, result.Error.Message)
+		d.Log.Error(err, "Error from Dynatrace provider")
+		return "", b, err
 	}
-
 	r := fmt.Sprintf("%f", d.getSingleValue(result))
 	return r, b, nil
 }
 
+func trimQuery(query string) string {
+	str := strings.ReplaceAll(query, " ", "")
```

Comment:
should we rather use query encoding to sanitize the query string? In the DT metrics API docs I found an example which does contain a whitespace when filtering for a certain entity name which might not work if we trim all whitespaces: https://www.dynatrace.com/support/help/dynatrace-api/environment-api/metric-v2/metric-selector#syntax-examples